SPECIFICATION.md

```diff
@@ -4,43 +4,72 @@ Authors: [Justin Dosse](mailto:justin.dossey@pathcheck.org) and [Vitor Pamplona]
-This document will use the following terms to define data types.
+## Data Types
 
+This document will use the following terms to define data types.
 1. **NUMERIC**: The **NUMERIC** data type is a sequence of integers between 0 and 99999999, inclusive.
-2. **STRING**: The **STRING** data type is a sequence of unicode characters encoded as UTF-8, up to 255 bytes after encoding. All Unicode strings MUST be [NFC Normalized](https://www.unicode.org/faq/normalization.html).
-3. **HASH**: The **HASH** data type is a sequence of alphanumeric characters containing a hexadecimal cryptographic hash. It is 64 bytes long.
-3. **SIGNATUREHEX**: The **SIGNATUREHEX** data type is a sequence of alphanumeric characters containing a hexadecimal cryptographic digest. It is up to 72 bytes long.
+2. **STRING**: The **STRING** data type is a sequence of UTF-8, [NFC Normalized](https://www.unicode.org/faq/normalization.html) characters, up to 255 bytes after encoding.
+3. **HASH**: The **HASH** data type is a sequence of 64 alphanumeric characters containing a hexadecimal hash.
+3. **SIGNATUREHEX**: The **SIGNATUREHEX** data type is a sequence of up-to-72 alphanumeric characters containing a hexadecimal digest.
 4. **DATE**: a date, in [ISO 8601 (YYYYMMDD) Basic Notation](https://en.wikipedia.org/wiki/ISO_8601). Example: `20200201` is 1 February, 2020.
 5. **SHORTSTRING**: a sequence of US-ASCII characters which is limited to 8 bytes in length.
 6. **SHORTNUMERIC**: a **NUMERIC** with a maximum value of 9.
 7. **PHONE**: a E.164 formatted phone number as string. US-ASCII, maximum 15 characters.
 
-## General Offline Credential Data Format
-All QR codes contain a message with a type, a version, a payload, and a cryptographic signature. The cryptographic signature is a SHA256 signature in hexadecimal form, calculated using the private ECDSA key of the **ISSUER**. The payload section is designated the **DATA** block and the cryptographic signature is contained in the **SIGNATURE** block. A block is an object containing a number of key-value pairs. The **type** field defines the message type and the **version** is a **NUMERIC** field defining the version of the type communicated in this QR code.
+# General Structure
+All QR codes contain a message with: 
+1. the **type** of certificate
+1. the **version**
+1. the **payload** itself
+1. the reference to a **public key**
+1. and a cryptographic **signature** of the payload
+
+The **type** field defines the payload type (`COUPON`, `PASSKEY`, `BADGE` or `STATUS`) and the **version** is a **NUMERIC** field defining the version of the type of certificate. The **payload** section contains the information related to the certificate itself. The cryptographic signature is a SHA256 signature in hexadecimal form, calculated using the private ECDSA key of the **ISSUER**. 
+
+The reference to the **public** key can be: 
+1. a URI to a DNS TXT Record containing the key for download.
+1. a URL address to download keys from. 
```

Comment:
I'm still not a fan of trying to fit a URL into keyId

Reply:
I am keeping it for now because it's very useful to test. But I also don't like it.